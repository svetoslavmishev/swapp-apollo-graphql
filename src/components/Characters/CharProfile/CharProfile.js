import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { useParams } from 'react-router-dom';
import { useQuery } from '@apollo/react-hooks';
import { useHistory } from 'react-router-dom';
import { GET_PROFILE } from '../../../queries/queries';
import { Card, CardMedia, CardContent, Typography } from '@material-ui/core';
import { Header } from '../../index';

const useStyles = makeStyles({
  card: {
    minWidth: 200,
    maxWidth: 300
  },
  bullet: {
    display: 'inline-block',
    margin: '0 2px',
    transform: 'scale(0.8)'
  },
  media: {
    height: 0,
    paddingTop: '75%'
  },
  title: {
    fontSize: 14
  },
  pos: {
    marginBottom: 12
  }
});

const CharProfile = () => {
  const classes = useStyles();
  const { characterId } = useParams();
  const history = useHistory();

  const { data, loading, error } = useQuery(GET_PROFILE, {
    variables: { id: characterId }
  });
  console.log(data);
  if (loading) return <div>Loading...</div>;
  if (error) return null;
  return (
    <>
      <Header />
      <div className={classes.root}>
        <Card className={classes.card}>
          <CardContent>
            <Typography variant="h5" component="h2">
              {data.person.name}
            </Typography>
            <CardMedia
              className={classes.media}
              image={data.person.image}
              title={data.person.name}
            />
            <Typography variant="body2" component="p">
              Height: {data.person.height}
            </Typography>
            <Typography variant="body2" component="p">
              Weight: {data.person.mass}
            </Typography>
            <Typography variant="body2" component="p">
              Species: {data.person.species.name}
            </Typography>
            <Typography variant="body2" component="p">
              Home World: {data.person.homeworld.name}
            </Typography>
          </CardContent>
        </Card>
        {data.person.starships.edges.map(ship => {
          return (
            <Card
              key={ship.node.id}
              onClick={() => history.push(`/starships/${ship.node.id}`)}
            >
              <CardContent
                style={{
                  display: 'flex',
                  maxWidth: 200,
                  borderRadius: 10,
                  alignItems: 'center',
                  margin: 6,
                  border: '1px solid red'
                }}
              >
                <CardMedia
                  style={{
                    width: 60,
                    height: 60,
                    borderRadius: 10
                  }}
                  image={ship.node.image}
                  title={ship.node.name}
                />
                <Typography variant="body2" component="p">
                  {ship.node.name}
                </Typography>
              </CardContent>
            </Card>
          );
        })}
      </div>
    </>
  );
};

export default CharProfile;
