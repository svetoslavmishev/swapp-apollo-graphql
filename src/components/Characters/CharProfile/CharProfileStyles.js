import { makeStyles } from '@material-ui/styles';

const styles = makeStyles(theme => ({
  root: {
    minHeight: 'calc(100vh - 64px)',
    background: ({ currentTheme }) =>
      theme.palette[currentTheme].backgroundColor
  },
  card: {
    minWidth: 180,
    maxWidth: 260,
    margin: 10,
    background: ({ currentTheme }) =>
      theme.palette[currentTheme].Cards.backgroundColor,
    color: ({ currentTheme }) => theme.palette[currentTheme].Cards.fontColor,
    border: ({ currentTheme }) =>
      `1px solid ${theme.palette[currentTheme].Cards.borderColor}`
  },
  title: {
    fontFamily: 'StarWars',
    color: ({ currentTheme }) =>
      theme.palette[currentTheme].PrimaryHeading.fontColor
  },
  media: {
    height: 0,
    paddingTop: '75%'
  },
  shipsTitle: {
    fontFamily: 'StarWars',
    textAlign: 'center',
    fontSize: 20,
    margin: 10,
    color: ({ currentTheme }) =>
      theme.palette[currentTheme].PrimaryHeading.fontColor,
    borderBottom: ({ currentTheme }) =>
      `1px solid ${theme.palette[currentTheme].Cards.borderColor}`
  }
}));

export default styles;
