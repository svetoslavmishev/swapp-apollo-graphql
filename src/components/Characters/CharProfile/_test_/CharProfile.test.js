import React from 'react';
import { shallow } from 'enzyme';
import CharProfile from '../CharProfile';
import { ApolloProvider } from '@apollo/react-hooks';

describe('<CharProfile />', () => {
  it('renders <CharProfile /> component without error', () => {
    const wrapper = shallow(
      <ApolloProvider>
        <CharProfile />
      </ApolloProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
