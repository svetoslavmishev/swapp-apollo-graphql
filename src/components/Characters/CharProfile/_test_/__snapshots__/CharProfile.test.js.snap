// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharProfile /> renders <CharProfile /> component without error 1`] = `ShallowWrapper {}`;
